### PR TITLE
DE conversion to seurat env

### PR DESCRIPTION
**Pull request by JuanluOnieva-biomage**
# Background
#### Link to issue 

Tickets: https://biomage.atlassian.net/browse/BIOMAGE-437?atlOrigin=eyJpIjoiY2ZkZGFhNjk4NWQ0NDk1NWJkYTE2MDg3NGViNjhmMDUiLCJwIjoiaiJ9

#### Links to any Pull Requests related to this

PR related: https://github.com/biomage-ltd/data-ingest/pull/4

#### Anything else the reviewers should know about the changes here

I have kept the same DE output format as pagoda2, but we need to discuss which columns we will show from it. 

# Changes
### Code changes

- Uploading Seurat instead of pagoda2
- Added functions related with DE for seurat object

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [x] Relevant Github READMEs updated
- [x] Relevant wiki pages created/updated

### Approvers
- [x] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

**Comment by iguanaonmystack** [comment]
I can't comment on the bioinformatics but good call on factoring this out into a new file.

**Review by marcellp — CHANGES_REQUESTED** [review]
This MR must not be merged until the data ingest scripts produce Seurat-compatible RDS objects.

File: r/Dockerfile
Comment (ivababukova) [inline r/Dockerfile]: if this is no longer going to be used, just delete it.

Comment (JuanluOnieva-biomage) [reply r/Dockerfile]: I have just pushed it again

**Comment by iguanaonmystack** [comment]
This looks fine to me. I think @marcellp is blocking this just to prevent it being accidentally merged?

**Comment by ivababukova** [comment]
that is correct

File: r/src/differential_expression.r
Comment (ogibson) [inline r/src/differential_expression.r:5]: Sebastian's suggestion:
Change this function for one implementation in R packages. 

E.g (https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/p.adjust)

**Review by marcellp — COMMENTED** [review]
This can now be merged in after approval from another reviewer.

**Review by marcellp — DISMISSED** [review]

**Review by ogibson — COMMENTED** [review]
I tried testing the DE script pulling the branch to my local environment but it throws some errors and didn't show differential expression correctly. Perhaps we should look into it a bit more.

File: r/src/differential_expression.r
Comment (JuanluOnieva-biomage) [reply r/src/differential_expression.r:5]: Thanks! I will measure the speed of both functions :). The reason I used this funcitonis  because is the one that is included in the pagoda2 DE pipeline.

**Review by ogibson — APPROVED** [review]
Tested it locally and read the code. Besides the bh.adjust function which is a kinda subjective issue, I think it's ok so let's merge!

**Comment by marcellp** [comment]
As Juanlu said he would be happy to merge it in if it is approved, I will do so right now.

**Pull request merged**
